pad.c

```diff
@@ -2844,6 +2844,91 @@ Perl_padname_dup(pTHX_ PADNAME *src, CLONE_PARAMS *param)
 
 #endif /* USE_ITHREADS */
 
+/*
+=for apidoc_section $lexer
+=for apidoc suspend_compcv
+
+Implements part of the concept of a "suspended complication CV", which can be
+used to pause the parser and compiler during parsing a CV in order to come
+back to it later on.
+
+This function saves the current state of the subroutine under compilation
+(C<PL_compcv>) into the supplied buffer.  This should be used initially to
+create the state in the buffer, as the final thing before a C<LEAVE> within a
+block.
+
+    ENTER;
+    start_subparse(0);
+    ...
+
+    suspend_compcv(&buffer);
+    LEAVE;
+
+Once suspended, the C<resume_compcv> or C<resume_compcv_and_save> function can
+later be used to continue the parsing from the point this stopped.
+
+=cut
+*/
+
+void
+Perl_suspend_compcv(pTHX_ struct suspended_compcv *buffer)
+{
+    PERL_ARGS_ASSERT_SUSPEND_COMPCV;
+
+    buffer->compcv = PL_compcv;
+
+    buffer->padix             = PL_padix;
+    buffer->constpadix        = PL_constpadix;
+
+    buffer->comppad_name_fill = PL_comppad_name_fill;
+    buffer->min_intro_pending = PL_min_intro_pending;
+    buffer->max_intro_pending = PL_max_intro_pending;
+
+    buffer->cv_has_eval       = PL_cv_has_eval;
+    buffer->pad_reset_pending = PL_pad_reset_pending;
+}
+
+/*
+=for apidoc A|void|resume_compcv_final|struct suspended_compcv *buffer
+
+Resumes the parser state previously saved using the C<suspend_compcv> for a
+final time before being compiled into a full CV.  This should be used within
+an C<ENTER>/C<LEAVE> scoped pair.
+
+=for apidoc A|void|resume_compcv_and_save|struct suspended_compcv *buffer
+
+Resumes a buffer previously suspended by the C<suspend_compcv> function, in a
+way that will be re-suspended at the end of the scope so it can be used again
+later.  This should be used within an C<ENTER>/C<LEAVE> scoped pair.
```

Comment:
```
./miniperl -Ilib autodoc.pl
Function 'resume_compcv_and_save', documented in pad.c, not listed in embed.fnc at autodoc.pl line 1819.
Function 'resume_compcv_final', documented in pad.c, not listed in embed.fnc at autodoc.pl line 1819.
```

Presumably these need m entries in embed.fnc

Reply:
But if I do that then it generates new lines in proto.h:

```
+/* PERL_CALLCONV void
+resume_compcv_and_save(pTHX_ struct suspended_compcv *buffer); */
+#define PERL_ARGS_ASSERT_RESUME_COMPCV_AND_SAVE
+
+/* PERL_CALLCONV void
+resume_compcv_final(pTHX_ struct suspended_compcv *buffer); */
+#define PERL_ARGS_ASSERT_RESUME_COMPCV_FINAL
+
```

Admittedly those function prototypes are commented out, but the assert macros remain.

Reply:
Ah actually I think that's all working now.